workspace/solution.go

```diff
@@ -87,3 +95,23 @@ func (s *Solution) PathToParent() string {
 	}
 	return filepath.Join(dir, s.Track)
 }
+
+func createIgnoreSubdir(path string) (string, error) {
+	path = filepath.Join(path, IgnoreSubdir)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if err := os.Mkdir(path, os.FileMode(0755)); err != nil {
+			return "", fmt.Errorf("failed to create directory: %s", path)
+		}
+	}
+	return path, nil
+}
+
+func migrateLegacySolutionFile(legacySolutionPath string, solutionPath string) error {
+	if _, err := createIgnoreSubdir(filepath.Dir(legacySolutionPath)); err != nil {
+		return err
+	}
+	if err := os.Rename(legacySolutionPath, solutionPath); err != nil {
+		return errors.New("failed migrating legacy solution file")
```

Comment:
This doesn't actually end up getting handled (it gets called in the path traversal loop in `workspace.SolutionDir`). If `os.Rename` fails, path traversal will continue until a [missing metadata error](https://github.com/exercism/cli/commit/53bb62de80806f504109901e95e931ea956884df) is reached.

Would an acceptable solution be to `panic`?  It seems that `os.Rename` really shouldn't fail and would be truly exceptional if it did. On the other hand, the "missing metadata error" is more useful to the user and provides an actionable solution (download a new exercise and copy files over).

Reply:
IMO the missing metadata message informs the user on how to resolve. If we panic with this error message the user has no where to go. 

Error messages with context and possible remediation steps is the way to go.